PolyEngine/Engine/Src/World.cpp

```diff
@@ -45,6 +45,14 @@ void World::DestroyEntity(const UniqueID& entityId)
 {
 	Entity* ent = IDToEntityMap[entityId];
 	HEAVY_ASSERTE(ent, "Invalid entity ID");
+
+	TransformComponent* transform = ent->GetComponent<TransformComponent>();
+	if (transform)
+	{
+		for (auto it : transform->GetChildren()) DeferredTaskSystem::DestroyEntityImmediate(this, it->GetOwnerID());
+		transform->GetChildren().Clear();
```

Comment:
No need to clear that collection. It will be empty by the time all children are removed.